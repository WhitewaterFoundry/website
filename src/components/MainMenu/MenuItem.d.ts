interface MenuItem {
  title: string
  link: string
}
export default MenuItem
