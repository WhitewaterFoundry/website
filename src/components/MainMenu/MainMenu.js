import React from "react"
import { Link } from "gatsby"
import styled from "styled-components"

const MainMenuWrapper = styled.div`
  width: 100%;
  height: 100%;
  display: grid;
  grid-template-columns: 70px auto;
  padding-right: 24px;
`

const MenuLinksWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
  width: 100%;
  height: 100%;
`

const MainMenuLink = styled(Link)`
  text-decoration: none;
  margin-left: 6px;
  margin-right: 6px;
`

const MainMenu = () => {
  return (
    <MainMenuWrapper>
      <div style={{ width: "70px" }}>{/* Icon Placeholder */}</div>
      <MenuLinksWrapper>
        <MainMenuLink to="/">Pengwin</MainMenuLink>
        <MainMenuLink to="/pengwin-enterprise">Pengwin Enterprise</MainMenuLink>
        <MainMenuLink to="/fedora-remix-for-wsl">Fedora Remix for WSL</MainMenuLink>
        <MainMenuLink to="/what-is-wsl">What is WSL?</MainMenuLink>
        <MainMenuLink to="/blog">Blog</MainMenuLink>
      </MenuLinksWrapper>
    </MainMenuWrapper>
  )
}

export default MainMenu
