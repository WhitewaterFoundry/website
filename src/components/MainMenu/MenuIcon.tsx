import React from "react"
import styled from "styled-components"

const Rect: React.FC<{
  className: string
  x: string
  y: string
  width: string
  height: string
}> = props => <rect {...props} x="4" height="2" fill="#363636" rx="1" />

const ButtonWrapper = styled.button`
  background: none;
  border: none;
  margin: 0;
  padding: 0;
  width: 48px;
  height: 48px;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  vertical-align: middle;
`

const IconWrapper = styled.div`
  .line {
    transition: all 150ms ease-in;
    transform-origin: 50% 50%;
  }

  .line-end {
    width: 38px;
  }

  .line-x {
    transform: translateX(-14px) translateY(2px) rotate(45deg);
    width: 51px;
  }

  .line-y {
    transform: translateX(-18px) translateY(-2px) rotate(-45deg);
    width: 51px;
  }

  .line-z {
    width: 0;
    transform: translateX(20px);
  }
`

const MenuIcon: React.FC<{ menuVisible: boolean; clickHandler: any }> = ({
  menuVisible,
  clickHandler,
}) => {
  return (
    <ButtonWrapper onClick={clickHandler}>
      <IconWrapper>
        <svg id="icon" width="40" height="40">
          <Rect
            className={`line ${menuVisible ? "line-x" : ""}`}
            x="4"
            y="10"
            width="34"
            height="2"
          />
          <Rect
            className={`line ${menuVisible ? "line-z" : ""}`}
            x="4"
            y="21"
            width="28"
            height="2"
          />
          <Rect
            className={`line ${menuVisible ? "line-y" : ""}`}
            x="4"
            y="34"
            width="34"
            height="2"
          />
        </svg>
      </IconWrapper>
    </ButtonWrapper>
  )
}

export default MenuIcon
