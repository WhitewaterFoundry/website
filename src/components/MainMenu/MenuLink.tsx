import styled from "styled-components"
import { Link } from "gatsby"

const MenuLink = styled(Link)`
  text-decoration: none;
  margin-left: 6px;
  margin-right: 6px;
`

export default MenuLink
