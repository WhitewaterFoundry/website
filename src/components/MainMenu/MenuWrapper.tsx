import styled from "styled-components"

const MenuWrapper = styled.div`
  width: 100%;
  height: 100%;
  display: grid;
  grid-template-columns: 70px auto;
  padding: 8px;
  background-color: #ffffff;
`

export default MenuWrapper
