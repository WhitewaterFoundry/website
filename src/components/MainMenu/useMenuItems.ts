import { useStaticQuery, graphql } from "gatsby"
import MenuItem from "./MenuItem"

type MenuItemsQueryData = {
  allMenuItemsJson: {
    edges: {
      node: MenuItem
    }[]
  }
}

const getMenuItems = (data: MenuItemsQueryData) =>
  data.allMenuItemsJson.edges.map(edge => edge.node)

export const useMenuItems = () => {
  const data = useStaticQuery(graphql`
    query menuItemsQuery {
      allMenuItemsJson {
        edges {
          node {
            link
            title
            id
          }
        }
      }
    }
  `)

  return getMenuItems(data)
}
