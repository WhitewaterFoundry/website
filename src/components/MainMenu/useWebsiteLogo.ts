import { useStaticQuery, graphql } from "gatsby"

export const useWebsiteLogo = () => {
  const data = useStaticQuery(graphql`
    {
      placeholderImage: file(relativePath: { regex: "/website-logo.png/" }) {
        childImageSharp {
          fluid(quality: 50) {
            ...GatsbyImageSharpFluid
          }
        }
      }
    }
  `)

  return data.placeholderImage.childImageSharp.fluid
}
