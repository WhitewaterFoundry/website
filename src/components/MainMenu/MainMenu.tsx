import * as React from "react"
import styled from "styled-components"
import MenuItem from "./MenuItem"
import MenuLink from "./MenuLink"

const MainMenuWrapper = styled.div`
  width: 100%;
  height: 100%;
  display: grid;
  grid-template-columns: 70px auto;
  padding-right: 24px;
`

const MenuLinksWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
  width: 100%;
  height: 100%;
`
interface Props {
  menuItems: Array<MenuItem>
  renderIcon: Function
}
const MainMenu: React.FC<Props> = ({ menuItems, renderIcon }) => {
  return (
    <MainMenuWrapper>
      <div style={{ width: "70px" }}>{renderIcon()}</div>
      <MenuLinksWrapper>
        {menuItems.map((item: MenuItem) => {
          return (
            <MenuLink to={item.link} key={item.id}>
              {item.title}
            </MenuLink>
          )
        })}
      </MenuLinksWrapper>
    </MainMenuWrapper>
  )
}

export default MainMenu
