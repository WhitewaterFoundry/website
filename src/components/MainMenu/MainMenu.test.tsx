import * as React from "react"
import { render, cleanup } from "react-testing-library"
import "jest-dom/extend-expect"
import MainMenu from "./MainMenu"
import MenuItem from "./MenuItem"
import "react-testing-library/cleanup-after-each"

const mockMenuItems: Array<MenuItem> = [
  { title: "Test 1", link: "/test1" },
  { title: "Test 2", link: "/test2" },
  { title: "Test 3", link: "/test3" },
  { title: "Test 4", link: "/test4" },
  { title: "Test 5", link: "/test5" },
]

const renderMainMenu: Function = () => {
  return render(<MainMenu menuItems={mockMenuItems} renderIcon={() => <div>Fake Icon</div>} />)
}

afterEach(cleanup)

describe("<MainMenu />", () => {
  it("Matches the previous accepted snapshot.", () => {
    const { asFragment } = renderMainMenu()
    expect(asFragment()).toMatchSnapshot()
  })

  it("Renders a link for each item passed to the menuItems prop in order.", () => {
    const { getAllByText } = renderMainMenu()

    let currentLinks = getAllByText(/^Test\s\d$/)

    mockMenuItems.forEach((item, i) => {
      expect(currentLinks[i].getAttribute("href")).toBe(item.link)
    })
  })

  it("Renders the element passed to the icon prop", () => {
    const { getByText } = renderMainMenu()
    const iconElement = getByText("Fake Icon")

    expect(iconElement).not.toBeUndefined()
  })
})
