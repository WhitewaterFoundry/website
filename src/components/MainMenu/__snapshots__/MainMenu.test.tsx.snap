// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainMenu /> Matches the previous accepted snapshot with mobile=false. 1`] = `
<DocumentFragment>
  <div
    class="sc-bwzfXH JrkCz"
  >
    <div
      style="width: 70px;"
    >
      <div>
        Fake Icon
      </div>
    </div>
    <div
      class="sc-ifAKCX ePtLCd"
    >
       
      <a
        class="sc-bdVaJa fthXlT"
        href="/test1"
      >
        Test 1
      </a>
      <a
        class="sc-bdVaJa fthXlT"
        href="/test2"
      >
        Test 2
      </a>
      <a
        class="sc-bdVaJa fthXlT"
        href="/test3"
      >
        Test 3
      </a>
      <a
        class="sc-bdVaJa fthXlT"
        href="/test4"
      >
        Test 4
      </a>
      <a
        class="sc-bdVaJa fthXlT"
        href="/test5"
      >
        Test 5
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<MainMenu /> Matches the previous accepted snapshot with mobile=true. 1`] = `
<DocumentFragment>
  <div
    class="sc-bwzfXH JrkCz"
  >
    <div
      style="width: 70px;"
    >
      <div>
        Fake Icon
      </div>
    </div>
    <div
      style="width: 100%; display: flex; flex-direction: row-reverse;"
    >
      <button
        class="sc-htpNat iTUDjS"
      >
        <div
          class="sc-bxivhb jpKkbv"
        >
          <svg
            height="40"
            id="icon"
            width="40"
          >
            <rect
              class="line "
              fill="#363636"
              height="2"
              rx="1"
              width="34"
              x="4"
              y="10"
            />
            <rect
              class="line "
              fill="#363636"
              height="2"
              rx="1"
              width="28"
              x="4"
              y="21"
            />
            <rect
              class="line "
              fill="#363636"
              height="2"
              rx="1"
              width="34"
              x="4"
              y="34"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      class="sc-EHOje hGQbOQ"
      height="0px"
      opacity="0"
    >
      <a
        class="sc-bdVaJa fthXlT"
        href="/test1"
      >
        Test 1
      </a>
      <a
        class="sc-bdVaJa fthXlT"
        href="/test2"
      >
        Test 2
      </a>
      <a
        class="sc-bdVaJa fthXlT"
        href="/test3"
      >
        Test 3
      </a>
      <a
        class="sc-bdVaJa fthXlT"
        href="/test4"
      >
        Test 4
      </a>
      <a
        class="sc-bdVaJa fthXlT"
        href="/test5"
      >
        Test 5
      </a>
    </div>
  </div>
</DocumentFragment>
`;
